client/roundCtrl.ts

```diff
@@ -23,19 +36,110 @@ import { Gating } from './gating';
 import { Promotion } from './promotion';
 import { pocketView, updatePockets, refreshPockets, Pockets } from './pocket';
 import { sound } from './sound';
-import { role2san, uci2cg, cg2uci, VARIANTS, IVariant, getPockets, getCounting, isHandicap, dropIsValid } from './chess';
+import {
+    role2san,
+    uci2cg,
+    cg2uci,
+    VARIANTS,
+    IVariant,
+    getPockets,
+    getCounting,
+    isHandicap,
+    dropIsValid,
+    DropOrig
+} from './chess';
 import { crosstableView } from './crosstable';
 import { chatMessage, chatView } from './chat';
 import { createMovelistButtons, updateMovelist, updateResult, selectMove } from './movelist';
 import { renderRdiff } from './profile'
 import { player } from './player';
 import { updateCount, updatePoint } from './info';
 import { notify } from './notification';
+import {
+    CrossTable,
+    MsgBoard,
+    MsgChat, MsgCtable,
+    MsgFullChat,
+    MsgGameNotFound,
+    MsgShutdown,
+    MsgSpectators,
+    MsgUserConnected,
+    Step
+} from "./analysisCtrl";
+import { PyChessModel } from "./main";
 
 const patch = init([klass, attributes, properties, listeners]);
 
 let rang = false;
 
+type MsgMove = {// cannot be interface because canot be converted to an indexed type and JSONObject, which is used in doSend is such

```

Comment:
These `Msg` types should be moved to `message.ts` as well until further notices.